cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataHttpHandlerTest.java

```diff
@@ -809,6 +809,76 @@ public void testScopeQueryParam() throws Exception {
   }
 
   @Test
+  public void testSearchTargetType() throws Exception {
+    Id.Namespace namespace = Id.Namespace.from("testSearchTargetType");
+    createNamespace(namespace.getId());
+    deploy(AllProgramsApp.class, Constants.Gateway.API_VERSION_3_TOKEN, namespace.getId());
+
+    // Add metadata to app
+    Set<String> tags = ImmutableSet.of("utag1", "utag2");
+    Id.Application appId = Id.Application.from(namespace, AllProgramsApp.NAME);
+    addTags(appId, tags);
+
+    // Add metadata to stream
+    tags = ImmutableSet.of("utag11");
+    Id.Stream streamId = Id.Stream.from(namespace, AllProgramsApp.STREAM_NAME);
+    addTags(streamId, tags);
+
+    // Add metadata to dataset
+    tags = ImmutableSet.of("utag21");
+    Id.DatasetInstance datasetId = Id.DatasetInstance.from(namespace, AllProgramsApp.DATASET_NAME);
+    addTags(datasetId, tags);
+
+    // Search for single target type
+    Assert.assertEquals(ImmutableSet.of(new MetadataSearchResultRecord(appId)),
+                        searchMetadata(namespace, "utag*", MetadataSearchTargetType.APP));
+    Assert.assertEquals(ImmutableSet.of(new MetadataSearchResultRecord(datasetId)),
+                        searchMetadata(namespace, "utag*", MetadataSearchTargetType.DATASET));
+
+    // Search for multiple target types
+    Assert.assertEquals(ImmutableSet.of(
+                          new MetadataSearchResultRecord(datasetId),
+                          new MetadataSearchResultRecord(streamId)
+                        ),
+                        searchMetadata(namespace, "utag*",
+                                       ImmutableSet.of(
+                                         MetadataSearchTargetType.DATASET,
+                                         MetadataSearchTargetType.STREAM
+                                       )
+                        ));
+
+    Assert.assertEquals(ImmutableSet.of(
+                          new MetadataSearchResultRecord(datasetId),
```

Comment:
so this search doesn't have an the metadata itself? or does ``searchMetadata()`` strip it out?